src/test/java/uk/gov/ch/oauth/OAuth2StateHandlerTest.java

```diff
@@ -86,7 +86,7 @@ void oauth2DecodeState() {
     void oauth2DecodeHandlesExceptionAndLogs() {
         String encoded = "words.words.words.words.words";
         assertNull(stateHandler.oauth2DecodeState(encoded));
-        verify(mockLogger).error(matches("Could not decode OAuth state"), any(JOSEException.class));
+        verify(mockLogger).error(matches("Could not decode OAuth state"), any(ParseException.class));
```

Comment:
Please check commit 
https://github.com/companieshouse/oauth-signin-java-library/pull/15/commits/ee2405970dcafc7b31845013de25ffc696b6944e

How will these changes cause merge conflicts going forwards?

Reply:
I chose to make them completely generic becuase the test was that it handles exceptions not that it handles a specific type of exception.

Reply:
If Oauth is expecting Exception then you won't see the failure, so no need to change OAuth

Reply:
But it's good to verify for the specific exception so oauth might need to be updated

Reply:
We're over specifing the exception. Testing its specifically a parse exception is a test of how the JWEObject is constructed.